### PR TITLE
add winternitz OTS suite 5 integration tests, fix compilation errors

### PR DESCRIPTION
**Pull request by ValentinVaninetti**
## Summary
Implements 8 comprehensive integration tests for Winternitz One-Time Signatures (Suite 5) covering sign/verify operations, serialization, and edge cases.

### New Tests (8)
- Sign/verify operations for SHA256 and HASH160
- Index and type mismatch validation
- Signature serialization round-trip
- Invalid input rejection
- Message reconstruction from signature
- Batch key derivation validation

### Bug Fixes (Required for compilation)
The project had pre-existing compilation errors due to API changes in `StorageConfig` (changed from `Option<Secret<String>>` to `Option<String>`). Fixed:
- 6 test functions in `key_manager.rs` (all within `#[cfg(test)]` module)
- 1 helper function in `tests/utils/helper.rs`
- 1 example in `examples/create.rs`

**Note:** All fixes are in test/example code only, no production code modified.

## Known Limitation
Tests have race conditions when run in parallel due to RocksDB. Run with:
```bash
cargo test winternitz_tests:: --lib -- --test-threads=1

### CODE REVIEW

**Review by danielemiliogarcia — CHANGES_REQUESTED** [review]
@ValentinVaninetti i think that the compilation errors are due to local versions being outdated, can we give it another try with the  workspace and all the dependencies fully updated? thanks

File: src/tests/mod.rs [hunk 1]
```diff
@@ -5,3 +5,5 @@ pub mod nonces_test;
 pub mod partial_signature_test;
 pub mod utils;
 pub mod winternitz_test;
+
+
```
Comment (andrelo86) [inline src/tests/mod.rs]: Remove double blanks

**Comment by andrelo86** [comment]
@ValentinVaninetti Did you try to run tests sequentially? we can avoid parallel execution errors. CI has a parameter to handle that scenario.

File: src/tests/mod.rs (hunk 1 above)
Comment (ValentinVaninetti) [reply src/tests/mod.rs]: On it, will remove double blanks.

As for the tests, yes i did ran the tests sequentially on local several times

**Comment by danielemiliogarcia** [comment]
@ValentinVaninetti do we need the changes at helper.rs? that might broke many other components using the helper

**Comment by ValentinVaninetti** [comment]
> @ValentinVaninetti do we need the changes at helper.rs? that might broke many other components using the helper

I can rollback the changes and probably modify all the tests, but from my side i needed the changes.
Can you provide me the components that might broke with this changes so i can review on my side?

File: src/tests/utils/helper.rs [hunk 2]
```diff
@@ -15,11 +15,14 @@ pub fn random_bytes() -> [u8; 32] {
 
 pub fn create_key_manager(
     store_keystore_path: &str,
-    encrypt: Option<Secret<String>>,
+    password: Option<String>,
 ) -> Result<KeyManager, anyhow::Error> {
-    let random_mnemonic: Mnemonic = Mnemonic::from_entropy(&random_bytes()).unwrap();
+    if let Some(parent) = std::path::Path::new(store_keystore_path).parent() {
+        std::fs::create_dir_all(parent).ok();
+    }
```
Comment (danielemiliogarcia) [inline src/tests/utils/helper.rs:22]: I understand if we need this (creating the dirs), and i don't mind doing the things in other way in the func body, but i will appreciate if we could left the signature as it was --> `encrypt: Option<Secret<String>>,`
I don't mind the parameter name, but the type.
for security the parameter should be a `Secret` if not that stacked value will not be secure deleted at drop, and may be left in memory, also the caller should be responsible of creating and managing the secret outside (before calling) so the function here is as clean as safe (that was required by the security audit)

**Review by jonasmartin — APPROVED** [review]
lgtm
